src/PsrLogAdapter.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB;
+
+use MongoDB\Driver\Monitoring\LogSubscriber;
+use MongoDB\Exception\UnexpectedValueException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use SplObjectStorage;
+
+use function MongoDB\Driver\Monitoring\addSubscriber;
+use function MongoDB\Driver\Monitoring\removeSubscriber;
+use function sprintf;
+
+/**
+ * Integrates libmongoc/PHPC logging with one or more PSR-3 loggers.
+ *
+ * This class is internal and should not be utilized by applications. Logging
+ * should be configured via the add_logger() and remove_logger() functions.
+ *
+ * @internal
+ */
+final class PsrLogAdapter implements LogSubscriber
+{
+    public const EMERGENCY = 0;
+    public const ALERT = 1;
+    public const CRITICAL = 2;
+    public const ERROR = 3;
+    public const WARN = 4;
+    public const NOTICE = 5;
+    public const INFO = 6;
+    public const DEBUG = 7;
+    public const TRACE = 8;
```

Comment:
Note: I removed the `LEVEL_` prefix on these constants, as it conflicted with the definitions on the LogSubscriber interface. PHP didn't seem to care about that, but Psalm did (noticed after correcting the extension version in the Psalm workflow).

These are for internal use only, so I don't particularly care about dropping the prefix. I also considered doing without these and relying on the PSR constants; however, we technically wouldn't have `TRACE` available (despite it being unlikely a spec would require its use).